pkgs/by-name/wa/waybar-lyric/package.nix

```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+buildGoModule (finalAttrs: {
+  pname = "waybar-lyric";
+  version = "0.10.0";
+
+  src = fetchFromGitHub {
+    owner = "Nadim147c";
+    repo = "waybar-lyric";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-glY0W/fB57pYrAhIxYSnkJBmqeN6vy0cwrS71CnKeK4=";
+  };
+
+  vendorHash = "sha256-DBtSC+ePl6dvHqB10FyeojnYoT3mmsWAnbs/lZLibl8=";
+
+  # Disable tests
+  # Tests fail with error "ERROR Failed to create cache directory", although program works as intended
+  doCheck = false;
```

Comment:
Testing locally, the checkPhase runs fine when enabled.

```suggestion
```

Note, if you do run into intermittent test failures, you can disable individual tests: https://nixos.org/manual/nixpkgs/unstable/#ssec-skip-go-tests

E.g.

```nix
  checkFlags =
    let
      # Skip tests that require network access
      skippedTests = [
        "TestNetwork"
        "TestDatabase/with_mysql" # exclude only the subtest
        "TestIntegration"
      ];
    in
    [
      "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$"
    ];
```

And as discussed previously, you can create writable directories using `mktemp -d` if absolutely necessary.